tests/fixtures/mod.rs

```diff
@@ -76,3 +81,85 @@ pub fn tmpdir() -> TempDir {
 pub fn port() -> u16 {
     free_local_port().expect("Couldn't find a free local port")
 }
+
+/// Run miniserve as a server; Start with a temporary directory, a free port and some
+/// optional arguments then wait for a while for the server setup to complete.
+#[fixture(args=&[] as &[&str])]
+#[allow(dead_code)]
+pub fn server<S>(args: impl IntoIterator<Item = S>) -> TestServer
+where
+    S: AsRef<std::ffi::OsStr>,
+{
+    let port = port();
+    let tmpdir = tmpdir();
+    let child = Command::cargo_bin("miniserve")
+        .expect("Couldn't find test binary")
+        .arg(tmpdir.path())
+        .arg("-p")
+        .arg(port.to_string())
+        .args(args)
+        .stdout(Stdio::null())
+        .spawn()
+        .expect("Couldn't run test binary");
+
+    sleep(Duration::from_secs(1));
+    TestServer::new(port, tmpdir, child)
+}
+
+/// Same as `server()` but ignore stderr
+#[fixture(args=&[] as &[&str])]
+#[allow(dead_code)]
+pub fn server_no_stderr<S>(args: impl IntoIterator<Item = S>) -> TestServer
+where
+    S: AsRef<std::ffi::OsStr>,
+{
+    let port = port();
+    let tmpdir = tmpdir();
+    let child = Command::cargo_bin("miniserve")
+        .expect("Couldn't find test binary")
+        .arg(tmpdir.path())
+        .arg("-p")
+        .arg(port.to_string())
+        .args(args)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()
+        .expect("Couldn't run test binary");
+
+    sleep(Duration::from_secs(1));
+    TestServer::new(port, tmpdir, child)
+}
+
+#[allow(dead_code)]
+pub struct TestServer {
+    port: u16,
+    tmpdir: TempDir,
+    child: Child,
+}
+
+#[allow(dead_code)]
+impl TestServer {
+    pub fn new(port: u16, tmpdir: TempDir, child: Child) -> Self {
+        Self {
+            port,
+            tmpdir,
+            child,
+        }
+    }
+    pub fn url(&self) -> Url {
+        Url::parse(&format!("http://localhost:{}", self.port)).unwrap()
+    }
+    pub fn path(&self) -> &std::path::Path {
+        self.tmpdir.path()
+    }
+    pub fn port(&self) -> u16 {
+        self.port
+    }
+}
+
+impl Drop for TestServer {
+    fn drop(&mut self) {
+        self.child.kill().expect("Couldn't kill test server");
+        /* TODO may need .wait() ? */
```

Comment:
Adding 
```
let ecode = self.child.wait()
                 .expect("failed to wait on child");

assert!(ecode.success());
```

should not harm. Rust docs say `On some systems, calling wait or similar is necessary for the OS to release resources. A process that terminated but has not been waited on is still around as a "zombie". Leaving too many zombies around may exhaust global resources (for example process IDs).` https://doc.rust-lang.org/std/process/struct.Child.html

Reply:
The "or similar" in the documentation confused me especially `.kill()` seems to leave no zombies on system! But I agree. A `.wait()` is reasonable.

Reply:
Let's resolve this. Can you switch it to `wait()`? If this fails too often, we'll kill instead.